### ISSUE TITLE
math/point/vec_norm: using vec_length is too expensive

### ISSUE THREAD

**Issue by rexim** [tsoding/nothing#657]
introduced in #652
  It involves multiplication and sqrt. We can just check if its components are close to 0.0f.

**Comment by ghost** [comment]
I don't think it can be improved that way because the length of the vector have to be calculated anyway in order to calculate the unit vector.
